hoomd/md/JanusFactor.h

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2009-2022 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+
+#ifndef __JANUS_FACTOR_H__
+#define __JANUS_FACTOR_H__
+
+#include "hoomd/HOOMDMath.h"
+#include "hoomd/VectorMath.h"
+
+#ifdef NVCC
+#define DEVICE __device__
+#else
+#define DEVICE
+#endif
+
+namespace hoomd
+    {
+namespace md
+    {
+/*! \file JanusFactor.h
+*/
+class JanusFactor
+{
+public:
+
+    struct param_type
+    {
+        param_type()
+            {
+            }
+
+        param_type(pybind11::dict params)
+            : cosalpha( fast::cos(params["alpha"].cast<Scalar>()) ),
+              omega(params["omega"].cast<Scalar>())
+            {
+            }
+
+        pybind11::dict asDict()
+            {
+                pybind11::dict v;
+
+                v["alpha"] = fast::acos(cosalpha);
+                v["omega"] = omega;
+
+                return v;
+            }
+
+        Scalar cosalpha;
+        Scalar omega;
+    }
+#ifdef SINGLE_PRECISION
+        __attribute__((aligned(8)));
+#else
+        __attribute__((aligned(16)));
+#endif
+
+    DEVICE JanusFactor(const Scalar3& _dr,
+                       const Scalar4& _qi,
+                       const Scalar4& _qj,
+                       const Scalar& _rcutsq,
+                       const param_type& _params)
+        : dr(_dr), qi(_qi), qj(_qj), params(_params)
+        {
+            // compute current janus direction vectors
+            vec3<Scalar> e { make_scalar3(1, 0, 0) };
```

Comment:
So did we decide that expanding this to allow for multiple patches was out of bounds for this PR?